website/source/guides/identity/approle-trusted-entities.html.md

```diff
@@ -0,0 +1,779 @@
+# TTL is set to 10 minutes, and Max TTL to be 30 minutes
+$ tee app-1-approle-role.json <<EOF
+{
+    "role_name": "app-1",
+    "bind_secret_id": true,
+    "secret_id_ttl": "10m",
+    "secret_id_num_uses": "1",
+    "token_ttl": "10m",
+    "token_max_ttl": "30m",
+    "period": 0,
+    "policies": [
+        "app-1-secret-read"
+    ]
+}
+EOF
+
+# AppRole backend configuration
+$ curl --silent \
+       --location \
+       --header "X-Vault-Token: $VAULT_TOKEN" \
+       --request POST \
+       --data @app-1-approle-role.json \
+       $VAULT_ADDR/v1/auth/approle/role/app-1
+```
+
+### Step 4: Configure Tokens for Terraform and Chef
+
+Now, you're ready to configure the policies and tokens to Terraform and Chef to
+interact with Vault. Remember, the point here is that you are giving each system
+a _limited_ token that is only able to pull either the `RoleID` or `SecretID`,
+_but not both_.
+
+![AppRole auth method workflow](/assets/images/vault-approle-tf-chef-2.png)
+
+#### Task 1: Create a policy and token for Terraform
+Create a token with appropriate policies allowing Terraform to pull
+the `RoleID` from Vault:
+
+```bash
+# Policy file granting to retrieve RoleID from Vault
+$ tee app-1-approle-roleid-get.hcl <<EOF
+{"policy":"path \"auth/approle/role/app-1/role-id\" {capabilities = [\"read\"]}"}
+EOF
+
+# Create the app-1-approle-roleid-get policy in Vault
+$ curl --silent \
+       --location \
+       --header "X-Vault-Token: $VAULT_TOKEN" \
+       --request PUT \
+       --data @app-1-approle-roleid-get.hcl \
+       $VAULT_ADDR/v1/sys/policy/app-1-approle-roleid-get
+
+# For Terraform
+# See: https://www.terraform.io/docs/providers/vault/index.html#token
+# Policy granting to create tokens required by Terraform
+$ tee terraform-token-create.hcl <<EOF
+{"policy":"path \"/auth/token/create\" {capabilities = [\"update\"]}"}
+EOF
+
+# Create the app-1-approle-roleid-get policy in Vault
+$ curl --silent \
+       --location \
+       --header "X-Vault-Token: $VAULT_TOKEN" \
+       --request PUT \
+       --data @terraform-token-create.hcl \
+       $VAULT_ADDR/v1/sys/policy/terraform-token-create
+
+# Payload to configure token for Terraform to pull RoleID
+$ tee roleid-token-config.json <<EOF
+{
+  "policies": [
+    "app-1-approle-roleid-get",
+    "terraform-token-create"
+  ],
+  "metadata": {
+    "user": "chef-demo"
+  },
+  "ttl": "720h",
+  "renewable": true
+}
+EOF
+
+# Get token and save it in roleid-token.json
+$ curl --silent \
+       --location \
+       --header "X-Vault-Token: $VAULT_TOKEN" \
+       --request POST \
+       --data @roleid-token-config.json \
+       $VAULT_ADDR/v1/auth/token/create > roleid-token.json
+```
+
+The token and associated metadata will be written out to the file
+`roleid-token.json`. The `client_token` value is what you'll give to Terraform.
+The file should look similar to the following:
+
+```plaintext
+$ cat roleid-token.json | jq
```

Comment:
Do we install jq for them in the script? If not we should make sure it's installed in that userdata script.

Reply:
I've already tested it all successfully, and `jq` is already on the image.